build/brigade.js

```diff
@@ -119,85 +138,36 @@ class GoJob extends Job {
   }
 }
 
-// checkRun is a GitHub Check Run that is ran as part of a Checks Suite,
-// running the provided runFunc corresponding to the provided description 
-function checkRun(e, p, runFunc, description) {
-  console.log(`Check requested: ${description}`);
-
-  // Create Notification object (which is just a Job to update GH using the Checks API)
-  note = new Notification(description.toLowerCase().replace(/[^a-z]/g, ''), e, p);
-  note.conclusion = "";
-  note.title = `Run ${description}`;
-  note.summary = `Running ${description} for ${e.revision.commit}`;
-  note.text = `Ensuring ${description} complete(s) successfully`
-
-  // Send notification, then run, then send pass/fail notification
-  return notificationWrap(runFunc(e, p), note);
+// runCheck is the default function invoked on a check_run:* event
+//
+// It determines which check is being requested (from the payload body)
+// and runs this particular check, or else throws an error if the check
+// is not found
+function runCheck(e, p) {
+  payload = JSON.parse(e.payload);
+
+  // Extract the check name
+  name = payload.body.check_run.name;
+
+  // Determine which check to run
+  switch (name) {
+    case `${projectName}-build`:
+      return runTests(e, p, build);
+    case `${projectName}-xbuild`:
+      return runTests(e, p, xbuild);
+    case `${projectName}-test`:
+      return runTests(e, p, test);
+    default:
+      throw new Error(`No check found with name: ${name}`);
+  }
 }
 
-// A GitHub Check Suite notification
-class Notification {
-  constructor(name, e, p) {
-    this.proj = p;
-    this.payload = e.payload;
-    this.name = name;
-    this.externalID = e.buildID;
-    this.detailsURL = `https://brigadecore.github.io/kashti/builds/${ e.buildID }`;
-    this.title = "running check";
-    this.text = "";
-    this.summary = "";
-
-    // count allows us to send the notification multiple times, with a distinct pod name
-    // each time.
-    this.count = 0;
-
-    // One of: "success", "failure", "neutral", "cancelled", or "timed_out".
-    this.conclusion = "neutral";
-  }
+// runTests is a Check Run that is run as part of a Checks Suite
+function runTests(e, p, jobFunc) {
+  console.log("Check requested");
 
-  // Send a new notification, and return a Promise<result>.
-  run() {
-    this.count++
-    var j = new Job(`${ this.name }-${ this.count }`, "deis/brigade-github-check-run:latest");
-    j.env = {
-      CHECK_CONCLUSION: this.conclusion,
-      CHECK_NAME: this.name,
-      CHECK_TITLE: this.title,
-      CHECK_PAYLOAD: this.payload,
-      CHECK_SUMMARY: this.summary,
-      CHECK_TEXT: this.text,
-      CHECK_DETAILS_URL: this.detailsURL,
-      CHECK_EXTERNAL_ID: this.externalID
-    }
-    return j.run();
-  }
+  var check = new Check(e, p, jobFunc(),
+    `https://brigadecore.github.io/kashti/builds/${e.buildID}`);
```

Comment:
Is this the right URL for our build?

Reply:
Good question - yes, the project is currently still using this Kashti dashboard.